ipaserver/plugins/cert.py

```diff
@@ -860,6 +864,27 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(csr_emails, principal_emails):
+    """
+    Checks if any email address from certificate does not
+    appear in ldap entry, comparing the domain part case-insensitively.
+    """
+
+    if not any(principal_emails):
+        return False
+
+    def lower_domain(email):
+        email_splited = email.split('@', 1)
+        email_splited[1] = email_splited[1].lower()
+
+        return '@'.join(email_splited)
```

Comment:
Yeah, I prefer to stay with the '@' join to be more consistent with the method name and all the context. If you think that it's not relevant at all, I can just remove it :)